pkg/cri/opts/spec_linux.go

```diff
@@ -448,6 +448,10 @@ func WithResources(resources *runtime.LinuxContainerResources, tolerateMissingHu
 		}
 		if limit != 0 {
 			s.Linux.Resources.Memory.Limit = &limit
+			// swap/memory limit should be equal to prevent container from swapping by default
+			if swapLimit == 0 {
+				s.Linux.Resources.Memory.Swap = &limit
+			}
 		}
 		if swapLimit != 0 {
 			s.Linux.Resources.Memory.Swap = &swapLimit
```

Comment:
maybe we should check `swapLimit` >= `limit` here?

Reply:
Good point. I assumed it was intentional to skip validation here based on the [discussion here](https://github.com/containerd/containerd/pull/6320/files#r761447063)?